server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -526,6 +527,8 @@ public Map<String, MappingMetadata> findMappings(
         Function<String, Predicate<String>> fieldFilter,
         Runnable onNextIndex
     ) {
+        assert Transports.assertNotTransportThread("decompressing mappings is too expensive for a transport thread");
```

Comment:
It looks like TransportGetAction does as well as a bunch of other ml actions (TransportOpenJobAction, TransportUpdateDatafeedAction, TransportRevertModelSnapshotAction, and TransportUpgradeJobModelSnapshotAction), right?. Do you want to remove this assertion from this PR to get this change in (since we've seen it causing problems) while we work out the rest?

Reply:
I think [9063af8](https://github.com/elastic/elasticsearch/pull/87679/commits/9063af8a05320a4ccd531e70dcbf951921189812) will fix these cases too, with #87911 tracking the removal of this tech debt.